common/src/main/java/com/microsoft/identity/common/internal/controllers/ExceptionAdapter.java

```diff
@@ -188,24 +198,143 @@ public static ServiceException getExceptionFromTokenErrorResponse(@NonNull final
         }
 
         outErr.setOauthSubErrorCode(errorResponse.getSubError());
+        setHttpResponseUsingTokenErrorResponse(outErr, errorResponse);
+        return outErr;
+    }
 
-        try {
-            outErr.setHttpResponse(
-                    synthesizeHttpResponse(
-                            errorResponse.getStatusCode(),
-                            errorResponse.getResponseHeadersJson(),
-                            errorResponse.getResponseBody()
-                    )
+
+    public static ServiceException getExceptionFromTokenErrorResponse(@Nullable final CommandParameters commandParameters,
+                                                                      @NonNull final TokenErrorResponse errorResponse) {
+
+        if(isIntunePolicyRequiredError(errorResponse)){
+            if(commandParameters == null || !(commandParameters instanceof BrokerSilentTokenCommandParameters)){
+                Logger.warn(TAG, "In order to properly construct the IntuneAppProtectionPolicyRequiredException we need the command parameters to be supplied.  Returning as service exception instead.");
```

Comment:
Just WARN?

Reply:
This shouldn't really happen... I suppose this is better as an error though... not ready to commit to throwing a full blow exception here though.

Reply:
Yeah.  Just something to draw somebody's eye when they're looking at logs.

Reply:
Since it means that whatever they were trying to do is about to fail, horribly.